drivers/pds/lib/utils.go

```diff
@@ -0,0 +1,578 @@
+	for i := 0; i < len(targetClusters); i++ {
+		if targetClusters[i].GetClusterId() == clusterID {
+			deploymentTargetID = targetClusters[i].GetId()
+			logrus.Infof("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
+		}
+	}
+	return tenantID, dnsZone, projectID, serviceType, deploymentTargetID
+}
+
+// GetClusterID retruns the cluster id for given targetClusterName
+func GetClusterID(projectID string, targetClusterName string) string {
+	deploymentTargets, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToProject(projectID)
+	if err != nil {
+		logrus.Fatalf("An Error Occured while listing deployment targets %v", err)
+	}
+	for index := range deploymentTargets {
+		if deploymentTargets[index].GetName() == targetClusterName {
+			return deploymentTargets[index].GetClusterId()
+		}
+	}
+	return ""
+}
+
+//GetStorageTemplate return the storage template id
+func GetStorageTemplate(tenantID string) string {
+	logrus.Infof("Get the storage template")
+	storageTemplates, _ := components.StorageSettingsTemplate.ListTemplates(tenantID)
+	for i := 0; i < len(storageTemplates); i++ {
+		if storageTemplates[i].GetName() == storageTemplateName {
+			logrus.Infof("Storage template details -----> Name %v,Repl %v , Fg %v , Fs %v",
+				storageTemplates[i].GetName(),
+				storageTemplates[i].GetRepl(),
+				storageTemplates[i].GetFg(),
+				storageTemplates[i].GetFs())
+			storageTemplateID = storageTemplates[i].GetId()
+			logrus.Infof("Storage Id: %v", storageTemplateID)
+		}
+	}
+	return storageTemplateID
+}
+
+// GetResourceTemplate get the resource template id and forms supported dataserviceNameIdMap
+func GetResourceTemplate(tenantID string, supportedDataServices []string) (map[string]string, map[string]string) {
+	logrus.Infof("Get the resource template for each data services")
+	resourceTemplates, _ := components.ResourceSettingsTemplate.ListTemplates(tenantID)
+	isavailable = false
+	for i := 0; i < len(resourceTemplates); i++ {
+		if resourceTemplates[i].GetName() == resourceTemplateName {
+			dataService, _ := components.DataService.GetDataService(resourceTemplates[i].GetDataServiceId())
+			for dataKey := range supportedDataServices {
+				if dataService.GetName() == supportedDataServices[dataKey] {
+					logrus.Infof("Data service name: %v", dataService.GetName())
+					logrus.Infof("Resource template details ---> Name %v, Id : %v ,DataServiceId %v , StorageReq %v , Memoryrequest %v",
+						resourceTemplates[i].GetName(),
+						resourceTemplates[i].GetId(),
+						resourceTemplates[i].GetDataServiceId(),
+						resourceTemplates[i].GetStorageRequest(),
+						resourceTemplates[i].GetMemoryRequest())
+
+					dataServiceDefaultResourceTemplateIDMap[dataService.GetName()] =
+						resourceTemplates[i].GetId()
+					dataServiceNameIDMap[dataService.GetName()] = dataService.GetId()
+					isavailable = true
+				}
+			}
+			isavailable = true
```

Comment:
can you have look at logic here. isavailalbe will be true?

Reply:
done.